packages/nx-monorepo/src/components/nx-project/index.ts

```diff
@@ -0,0 +1,304 @@
+  public addImplicitDependency(...dependee: (Project | string)[]) {
+    this.implicitDependencies.push(
+      ...dependee.map((_d) => (_d instanceof Project ? _d.name : _d))
+    );
+  }
+
+  /** Set `namedInputs` helper */
+  public setNamedInput(name: string, inputs: string[]): void {
+    this.namedInputs[name] = inputs;
+  }
+
+  /** @internal */
+  protected _getTargetDefaults(name: string): Nx.IProjectTarget | {} {
+    return NxWorkspace.of(this.project)?.targetDefaults[name] || {};
+  }
+
+  /** Set `targets` helper */
+  public setTarget(
+    name: string,
+    target: Nx.IProjectTarget,
+    includeDefaults: boolean | string = false
+  ): void {
+    let _default = {};
+    if (includeDefaults) {
+      if (this.targets[name]) {
+        _default = this.targets[name];
+      } else {
+        (_default = this._getTargetDefaults(
+          includeDefaults === true ? name : includeDefaults
+        )),
+          this.targets[name] || {};
+      }
+    }
+    this.targets[name] = deepMerge([_default, target], { append: true });
+  }
+
+  /**
+   * Add input and output files to build target
+   * @param inputs Input files
+   * @param outputs Output files
+   * @param excludeOutputs Indicates if output files are automatically excluded from input files
+   */
+  public addBuildTargetFiles(
+    inputs?: (string | Nx.IInput)[],
+    outputs?: string[],
+    excludeOutputs: boolean = false
+  ): void {
+    if (outputs && excludeOutputs) {
+      inputs = [
+        ...(inputs || []),
+        ...outputs.flatMap((o) => [`!${o}`, `!${o}/**/*`]),
+      ];
+    }
+
+    this.setTarget(
+      "build",
+      {
+        inputs: inputs || [],
+        outputs: outputs || [],
+      },
+      true
+    );
+  }
+
+  /** @interface */
+  synthesize() {
+    const projectPath = path.relative(
+      this.project.root.outdir,
+      this.project.outdir
+    );
+
+    const packageManager =
+      NodePackageUtils.tryFindNodePackage(this.project, true)?.packageManager ||
+      NodePackageManager.NPM;
+
+    this.project.tasks.all
+      .filter((task) => {
+        if (
+          this.includedScripts.length &&
+          !this.includedScripts.includes(task.name)
+        ) {
+          // Exclude tasks that are not in explicit "includeScripts" when defined
+          return false;
+        }
+        if (task.name in this.targets) {
+          // always include tasks that were explicitly added to nx targets
+          return true;
+        }
+        if (
+          NODE_LIFECYCLE_TASKS.includes(task.name) &&
+          NodePackageUtils.isNodeProject(this.project)
+        ) {
+          // exclude node lifecycle tasks for node based projects
+          return false;
+        }
+        return true;
+      })
+      .forEach((task) => {
+        const _target = this.targets[task.name] || {};
+        _target.executor = _target.executor || "nx:run-commands";
+        _target.options = {
+          command: NodePackageUtils.command.downloadExec(
```

Comment:
I copied it from the code you wrote before 😊 ... assuming was to do with non-node based package/repo support? Since before was only used for non-node projects.

Reply:
I have removed the `dlx` stuff - using `NodePackageUtils.command.projen(packageManager, task.name)` now